### DISCUSSION TITLE
Add automatic benchmark of storage for full proofs during farming

### DISCUSSION THREAD

**Discussion by marcoabreu** [Chia-Network/chia-blockchain#5064] [Ideas]
Right now, we are only notifying the user when the qualities lookup is too slow. In the rare case they are actually having a full proof though, they might run into the issue that their storage is not capable of handling the load and thus they might lose the reward.

Hence, my proposal would be run period automatic benchmarks of a random farmed plot which measures the duration of full proof and lets the user know if their storage is too slow. Thus, we could avoid the frustrating situation that a user has a bunch of plots but never actually receives the rewards, but they could never figure out why.

This pays into https://github.com/Chia-Network/chia-blockchain/discussions/3188

**Comment by marcoabreu** [comment]
Implemented in https://github.com/Chia-Network/chia-blockchain/pull/5170

**Comment by fiveangle** [comment]
Realistically, if a lookup goes beyond 30s all components should give up and return immediately with an ERROR condition trickled up the stack since even the most optimal proof of space would not be candidate since the SP will have expired.  Yet, I see responses in the logs of times double and even triple of that (when there are underlying data access issues).
